core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/diskModels.scala

```diff
@@ -104,4 +104,8 @@ object FormattedBy extends Enum[FormattedBy] {
   final case object Galaxy extends FormattedBy {
     override def asString: String = "GALAXY"
   }
+
+  final case object Custom extends FormattedBy {
```

Comment:
I wonder if it makes sense to have each supported app to be its own data type...cuz a disk formated by one app can't be re-used by another

Reply:
Hm.. the point of this is to not have to pre-define apps in Leo. I wouldn't want to have to make a code change for each new app (unless infosec says there needs to be an allow-list of apps, of course… but this doesn't seem to be the case).

I don't think it's necessarily true that a disk can't be re-used across app types. A disk with data on it populated with Jupyter can be plugged into cirrocumulus for example. I think it's just Galaxy disks that we want to ensure only get used with Galaxy, because they do some special NFS formatting.

Reply:
Yeah, I didn't want to enumerate the apps here for this reason. Thats why I talk a bit about how formattedBy could be derived from a descriptor in the description, and mention some other issues with it. I plan to make a spike to plan out how exactly formattedBy will work in the custom-apps world, [link incoming]